models/sessions/scratch/bigquery/snowplow_web_sessions_this_run.sql

```diff
@@ -54,13 +54,37 @@ with session_firsts as (
         mkt_campaign as mkt_campaign,
         mkt_clickid as mkt_clickid,
         mkt_network as mkt_network,
+        regexp_extract(page_url ,r'utm_source_platform=([^?&#]*)') as mkt_source_platform,
+        case
+            when mkt_source = '(direct)' and mkt_medium in ('(not set)', '(none)') then 'Direct'
+            when CONTAINS_SUBSTR(mkt_medium, 'cross-network') then 'Cross-network'
+            when REGEXP_CONTAINS(mkt_medium, r'^(.*cp.*|ppc|retargeting|paid.*)$') then
+            case
+                when mkt_source in ({{select_source_category('SOURCE_CATEGORY_SHOPPING')}}) or REGEXP_CONTAINS(mkt_campaign, r'^(.*(([^a-df-z]|^)shop|shopping).*)$)') then 'Paid Shopping'
+                when mkt_source in ({{select_source_category('SOURCE_CATEGORY_SEARCH')}}) then 'Paid Search'
```

Comment:
Would it make sense for this stuff to be a macro so that someone could overwrite it if they wanted to?